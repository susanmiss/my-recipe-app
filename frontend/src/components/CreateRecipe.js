import React, {useState} from 'react';
import axios from 'axios';




const CreateRecipe = () => {

    const [state, setState] = useState({
        recipeTitle: '',
        recipeContent: ''

    })


    const {recipeTitle, recipeContent} = state 

    const handleChange = (name) => (event) => {
        //console.log('name', name, 'event', event.target.value)
        setState({...state, [name]: event.target.value})
    }

    //HANDLE SUBMIT
    const handleSubmit = event => {
        event.preventDefault() 
        axios.post(`${process.env.REACT_APP_API}/create`, {recipeTitle, recipeContent})
        .then(response => {
            console.log(response)
            setState({...state, recipeTitle: '', recipeContent: ''});
            // setContent('');
            alert(`Recipe title ${response.data.title} is created.`)
        })
        .catch(error => {
            console.log(error.response)
            alert(error.response.data.error);
        })
    }


    return (
        <div className="container p-5">
          <br />
          <h1>Add Your Favorite Fat Recipe:</h1>
          <br />
          {/* {JSON.stringify(state)} */}
          <form onSubmit={handleSubmit}>

              <div className="form-group" >
                  <label className="text-muted">Recipe Title</label>
                  <input onChange={handleChange('recipeTitle')} value={recipeTitle} type="text" className="form-control" placeholder="Title Recipe" required />
              </div>

              <div className="form-group" >
                  <label className="text-muted">Recipe Title</label>
                  <textarea onChange={handleChange('recipeContent')} value={recipeContent} type="text" className="form-control txt-rec" placeholder="Recipe" required ></textarea>
              </div>
      
        
              <div>
                  <button className="primary">Create your Recipe</button>
              </div>
          </form>
        </div>
      )
}
export default CreateRecipe;
