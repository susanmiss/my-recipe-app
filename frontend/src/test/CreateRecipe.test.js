import React from 'react';
import ReactDOM from 'react-dom'
import {shallow} from 'enzyme'; 
import renderer from 'react-test-renderer';
import CreateRecipe from '../components/CreateRecipe'


describe("<CreateRecipe />", () => {
    
  it('mounts to DOM', () => {
      const div = document.createElement('div');
      ReactDOM.render(<CreateRecipe />, div);
      ReactDOM.unmountComponentAtNode(div);
  });
  it('renders correctly', () => {
      const tree = renderer
          .create(<CreateRecipe />)
          .toJSON();
      expect(tree).toMatchSnapshot();
  });
  let wrapper;
  beforeEach(() => wrapper = shallow(<CreateRecipe />));
  
  it('Renders correctly', () => {
      expect(wrapper).toMatchSnapshot();
  })
  
  it('should render a <div />', () => {
      expect(wrapper.find('div').length).toEqual(4);
  });

  it('should render 1 form', () =>{
    expect(wrapper.find('form').length).toEqual(1);
})
 
})