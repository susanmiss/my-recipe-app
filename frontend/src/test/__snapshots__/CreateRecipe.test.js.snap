// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CreateRecipe /> Renders correctly 1`] = `ShallowWrapper {}`;

exports[`<CreateRecipe /> renders correctly 1`] = `
<div
  className="container p-5"
>
  <br />
  <h1>
    Add Your Favorite Fat Recipe:
  </h1>
  <br />
  <form
    onSubmit={[Function]}
  >
    <div
      className="form-group"
    >
      <label
        className="text-muted"
      >
        Recipe Title
      </label>
      <input
        className="form-control"
        onChange={[Function]}
        placeholder="Title Recipe"
        required={true}
        type="text"
        value=""
      />
    </div>
    <div
      className="form-group"
    >
      <label
        className="text-muted"
      >
        Recipe Title
      </label>
      <textarea
        className="form-control txt-rec"
        onChange={[Function]}
        placeholder="Recipe"
        required={true}
        type="text"
        value=""
      />
    </div>
    <div>
      <button
        className="primary"
      >
        Create your Recipe
      </button>
    </div>
  </form>
</div>
`;
